pkg/yang/file.go

```diff
@@ -115,22 +124,43 @@ func findFile(name string) (string, string, error) {
 	return "", "", fmt.Errorf("no such file: %s", name)
 }
 
-// findInDir looks for a file named name in dir or any of its subdirectories.
-func findInDir(dir, name string) string {
+// findInDir looks for a file named name in dir or any of its subdirectories if
+// recurse is true. if recurse is false, scan only the directory dir.
+func findInDir(dir, name string, recurse bool) string {
 	fis, err := ioutil.ReadDir(dir)
 	if err != nil {
 		return ""
 	}
+	var candidates []string
+	mname := name
+	if strings.HasSuffix(mname, ".yang") {
+		mname = name[:len(name)-len(".yang")]
+	}
+
 	for _, fi := range fis {
 		if !fi.IsDir() {
 			if fi.Name() == name {
 				return filepath.Join(dir, name)
+			} else if !strings.Contains(name, "@") && strings.HasPrefix(fi.Name(), mname+"@") && strings.HasSuffix(fi.Name(), ".yang") {
```

Comment:
It's not implicit, I don't believe. The first check is against `name` (i.e., if the request is for a non "revisioned" filename) wile the second check is against `fi.Name()`, the name we're iterating over, and confirms that the file (possibly) has a revision and thus is an eligible candidate.

From RFC6020 [s5.2.](https://tools.ietf.org/html/rfc6020#section-5.2):
```
   YANG modules and submodules are typically stored in files, one module
   or submodule per file.  The name of the file SHOULD be of the form:

     module-or-submodule-name ['@' revision-date] ( '.yang' / '.yin' )

   YANG compilers can find imported modules and included submodules via
   this convention.
```

So we confirm that the filename we're iterating over starts with `module-name@` and ends in `.yang`, which (perhaps with the addition of a length check to validate the `revision-date` is not empty?) is enough to satisfy the constraint described above.